pkg/collector/corechecks/embed/apm.go

```diff
@@ -31,17 +32,49 @@ type apmCheckConf struct {
 
 // APMCheck keeps track of the running command
 type APMCheck struct {
-	cmd *exec.Cmd
+	binPath     string
+	commandOpts []string
+	running     uint32
+	stop        chan struct{}
+	stopDone    chan struct{}
 }
 
 func (c *APMCheck) String() string {
 	return "APM Agent"
 }
 
-// Run executes the check
+// Run executes the check with retries
 func (c *APMCheck) Run() error {
+	atomic.StoreUint32(&c.running, 1)
```

Comment:
`Run()` being exported, we should maybe ensure that if `c.running == 1` and we call again the function we return an error instead of launching again?

Reply:
I'd say the check's expectation is that, since its Interval is `0`, it won't ever be `Run` more than once until the `Run` returns or `Stop` is called ; and since `Stop` is blocking until the check is actually stopped, I don't think we need to check for that here.

Basically all I'm saying is that I don't think it's the check's responsibility to ensure that. It'd be an easy thing to add but I don't really want to check for things that shouldn't happen by design, if that makes sense :)

Anyway let me know what you think of this